app/models/api-articles.model.js

```diff
@@ -21,35 +24,49 @@ exports.selectArticles = () => {
     a.votes,
     a.article_img_url,
     COUNT(c.comment_id) AS comment_count
-FROM 
+    FROM 
     articles a
-LEFT JOIN 
-    comments c ON a.article_id = c.article_id
-GROUP BY 
-    a.author,
-    a.title,
-    a.article_id,
-    a.topic,
-    a.created_at,
-    a.votes,
-    a.article_img_url
+    LEFT JOIN 
+    comments c ON a.article_id = c.article_id`;
+
+    if (topic) {
+        await doesTopicExist(topic);
+        sqlQuery += ` WHERE a.topic = $1`;
+        queryValues.push(topic);
+    }
+
+  sqlQuery += ` GROUP BY 
+a.author,
+a.title,
+a.article_id,
+a.topic,
+a.created_at,
+a.votes,
+a.article_img_url
 ORDER BY 
-    a.created_at DESC;`)
-    .then((response) => {
-        return response.rows
-    })
-}
+a.created_at DESC;`;
+
+const response = await db.query(sqlQuery, queryValues);
+return response.rows;
+};
 
 exports.patchArticleById = (inc_votes, article_id) => {
-    if (isNaN(inc_votes) || !Number.isInteger(Number(inc_votes))){
-        return Promise.reject({status: 400, message: 'Bad Request: Invalid inc_votes data'})
-    }
+  if (isNaN(inc_votes) || !Number.isInteger(Number(inc_votes))) {
```

Comment:
One tweak i'd make to this condition is using the Number() constructor around inc_votes before checking weather it's NaN or not - this will be a true check for if this is a number, which it currently isn't. This is because NaN is a specific thing in javascript it **doesn't** mean 'anything that's not an object'

When we try to convert a variable to a Number, if it can't be done because there's no way to convert it to a number, then we'll get NaN which we can use to find out if inc_votes is a valid number